proposals/secure-storage-polyfill.js

```diff
@@ -0,0 +1,67 @@
+// Existing browser APIs don't give us access to the system's secure storage,
+// meaning all data in this polyfill is stored in less secure mechanisms.
+console.warn(
+  "Warning: browser.secureStorage polyfill loaded. This proof of concept stores data insecurely and should not be used in production."
+);
+
+const RECOGNISED_AUTH_METHODS = [
+  "PIN",
+  "PASSWORD",
+  "BIOMETRY_FACE",
+  "BIOMETRY_FINGERPRINT",
+];
+
+window.secureStorage = {
+  getInfo: () => {
+    return {
+      type: "MACOS_KEYCHAIN",
+      availableAuthentication: RECOGNISED_AUTH_METHODS,
+    };
+  },
+  store: (request) => {
```

Comment:
all those methods should probably be async

Reply:
Definitely a good call, I've added the async keyword for each of these.